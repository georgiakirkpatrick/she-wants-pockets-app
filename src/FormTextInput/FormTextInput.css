.FormText {
    width: 100%;
}

.FormText > input {
    display: block;
    width: 100%;
    margin: 2px 0px 10px;
    padding: 10px;
    font: inherit;
}