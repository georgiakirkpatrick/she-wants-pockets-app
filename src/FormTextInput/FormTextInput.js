import React from 'react'
import './FormTextInput.css'

const FormTextInput = props => {
    return (
        <div className='FormText'>
            <label htmlFor={props.id}>{props.prompt}</label>
            <input 
                type='text' 
                id={props.id} 
                name={props.name} 
                defaultValue={props.currentValue} 
                onChange={props.handleChange} 
            />
        </div>
    )
}

FormTextInput.defaultProps = {
    id: 1,
    prompt: '',
    name: '',
    currentValue: 1,
    change: () => {}
}

export default FormTextInput