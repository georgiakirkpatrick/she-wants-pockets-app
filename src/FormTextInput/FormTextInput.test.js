import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import FormTextInput from './FormTextInput'

describe('FormTextInput', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FormTextInput />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<FormTextInput />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})