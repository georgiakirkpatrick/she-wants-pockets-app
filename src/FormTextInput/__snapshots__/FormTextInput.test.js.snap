// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput render the UI as expected 1`] = `
<div
  className="FormText"
>
  <label
    htmlFor={1}
  >
    
  </label>
  <input
    defaultValue={1}
    id={1}
    name=""
    type="text"
  />
</div>
`;
